### PR TITLE
Use OVS bridges on KVM hypervisors in Cosmic

### PR DESCRIPTION
**Pull request by remibergsma**
This switches the KVM hypervisors used by Cosmic from `LinuxBridges` to `OpenvSwitch bridges`. 

Using the override settings, the `firstboot` script used by KVM on Cosmic is now the `OpenvSwitch` enabled script, instead of the defualt script. CloudStack still uses LinuxBridges as I didn't get that to work with ovs.

I've successfully tested deploying a Cloud with the new KVM OVS enabled hypervisors.

Ping @borisroman could you please review and help testing this on the Jenkins slaves? I think I made the necessary changes but wasn't able to test it.

Ping @neubauerf does this look good to you?

### CODE REVIEW

**Comment by remibergsma** [comment]
image

image

**Comment by remibergsma** [comment]
OVS `2.5.1` is installed by default, which is `STT` enabled. Should we want to hook up NSX, then we don't need to change the hypervisors.

**Comment by sspans** [comment]
We don't currently publish these ovs packages, so this would make it harder for others to deploy a bubble.

**Comment by remibergsma** [comment]
@sspans There is a setting that can be flipped to cloudstack in order to get the old behaviour so I think we're fine. OVS and KVM doesn't work well with CloudStack, not even with the community 2.4.x packages. 

I could add some code to see if we can get out mctadm packages and otherwise fallback to community. That way people can spin cosmic without our packages so that would make sense.

**Comment by remibergsma** [comment]
@sspans I could add some code to see if we can get out mctadm packages and otherwise fallback to community. That way people can spin cosmic without our packages so that would make sense.

**Comment by remibergsma** [comment]
@sspans Thanks for the suggestion, improved it. Now works both at SBP and in community. By default we run KVM with OVS and vlans. Optionally NSX can be used as well. STT only works in SBP version.

image

File: ci/ci-deploy-infra.sh [hunk 1]
```diff
@@ -38,21 +38,25 @@ function install_kvm_packages {
   scp_base="sshpass -p ${hvpass} scp -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o LogLevel=quiet "
 
   # scp packages to hypervisor, remove existing, then install new ones
-  ${ssh_base} ${hvuser}@${hvip} rm -f cloudstack-\*
```
Comment (miguelaferreira) [inline ci/ci-deploy-infra.sh:41]: Is this script used for cosmic or cloudstack CI?

Comment (remibergsma) [inline ci/ci-deploy-infra.sh:41]: All scripts in `ci/` folder are used by Jenkins for Cosmic only.

Comment (remibergsma) [inline ci/ci-deploy-infra.sh:41]: As discussed, apparently that's not true :-) We'll clean up the old cloudstack scripts later.

**Comment by remibergsma** [comment]
Rebased and resolved conflict due to deleted scripts.

**Comment by bheuvel** [comment]
As this was used along with https://github.com/MissionCriticalCloud/cosmic-plugin-hypervisor-kvm/pull/30 , resulting in all green balls, LGTM :+1:

**Pull request merged**